README.asciidoc

```diff
@@ -26,9 +26,157 @@ If you prefer to install and manage Elasticsearch yourself, you can download
 the latest version from 
 https://www.elastic.co/downloads/elasticsearch[elastic.co/downloads/elasticsearch].
 
-For more installation options, see the
-https://www.elastic.co/guide/en/elasticsearch/reference/current/install-elasticsearch.html[Elasticsearch installation
-documentation].
+=== Set up a local dev deployment
+
+//// 
+IMPORTANT: This content is replicated in the Elasticsearch guide. 
+If you make changes, you must also update setup/set-up-local-dev-deployment.asciidoc.
+////
+
+To try out Elasticsearch on your dev machine, we recommend using Docker
+and running both Elasticsearch and Kibana.
+Docker images are available from the https://www.docker.elastic.co[Elastic Docker registry].
+
+For other installation options, see the
+https://www.elastic.co/guide/en/elasticsearch/reference/current/install-elasticsearch.html[Elasticsearch installation documentation].
+
+**Install and run Elasticsearch**
+
+. Install and start https://www.docker.com/products/docker-desktop[Docker
+Desktop]. Go to **Preferences > Resources > Advanced** and set Memory to at least 4GB.
+
+. Start an Elasticsearch container:
++
+----
+docker network create elastic
+docker pull docker.elastic.co/elasticsearch/elasticsearch:{version} <1>
+docker run --name es-node1 --net elastic -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" -t docker.elastic.co/elasticsearch/elasticsearch:{version}
+----
+<1> Replace {version} with the version of Elasticsearch you want to run, such as 8.4.1.
```

Comment:
This is an example of where having `latest` tags for this stuff would be really helpful. We really want to avoid having hard-coded version numbers in the README. Unlike our docs, we can't use variable substitution to ensure this is always the latest release. We'd have to constantly keep this up to date.

Reply:
Agree it would be really helpful! For now, I deliberately did not hard code the version in the snippet and specified a version as an example in the callout. It's not ideal, but I it makes it less critical to bump the version every release.

Reply:
I think I'd avoid having an example version here at all as it'll quickly be out of date and we don't want to encourage folks to use old versions.

Reply:
@elasticmachine run elasticsearch-ci/docs